components/theia/packages/gitpod-extension/src/browser/gitpod-user-storage-provider.ts

```diff
@@ -28,6 +28,11 @@ export class GitpodUserStorageProvider implements FileSystemProviderWithFileRead
     readonly capabilities = FileSystemProviderCapabilities.FileReadWrite;
     readonly onDidChangeCapabilities = Event.None;
 
+    @postConstruct()
+    initialize() {
+        this.startPeriodicCacheWriter(10);
+    }
+
     // TODO server could provide stats instead
     protected readonly stats = new Map<string, Stat>();
```

Comment:
It looks like `stats` should be a part of `contentCache` value.

Reply:
I saw the similarity but felt they are nicely decoupled atm.

Reply:
I think in `readFile` it gets unnecessary updated now without actual content change, i.e. stats should be updated then a value in `contentCache` is changed.

Reply:
I'm not 100% sure what you mean. Looking at the code `stat` is always updated "one tick after" `content`...? Or should we update it before writing to `server`?

I tried to do https://github.com/gitpod-io/gitpod/pull/2476/commits/ab19d7762354b3ae36f2fab9749b3b5aac91122d, but that was the case before, no?

Reply:
I wanted that mtime should be updated each time when a new value is written to contentCache and ctime when first value is written. I think it does know it.